### ISSUE TITLE
Build with debugging symbols on CI runs

### ISSUE THREAD

**Issue by jrs65** [kotekan/kotekan#719]
This would do a few things:
- Reduce the object size, and allow us to store more in the ccache
- Improve link times (because of the reduced binary size)
- Free up disk space on the Github environment for tests

I don't think there's any downsides. It's not like we can go and run gdb against a build anyway.

**Comment by andrerenard** [comment]
I agree, the only things to be careful about are `asserts` and `DEBUG` statements in the unit tests.  Although maybe we can build a reduced version just for unit tests which don't require as many optional features (i.e. GPUs)?
